tests/sgl/core/test_logger.cpp

```
@@ -0,0 +1,50 @@
+
+    Logger* m_logger;
+    LogLevel m_level{LogLevel::none};
+    std::string m_module;
+    std::string m_msg;
+    std::string m_logger_name;
+    size_t m_write_count{0};
+};
+
+TEST_CASE("output callback can re-enter logger")
+{
+    auto logger = Logger::create(LogLevel::info, "test", false);
```

Cppcheck (2.21.0)

[style] The function 'class_name' is never used.

(unusedFunction)